content/blog/2018/07/2018-07-12-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,190 @@
+* Build specifications for pull request can be written in a concise declarative format.
+* Build reports will be automatically published to Github, Bitbucket, etc.
+* Build status updates will be sent to git servers automatically.
+* Users will not have to deal with pipeline code.
+* If there will be no merge conflicts or build failures, the PR can be merged into target branch.
+
+=== Phase 1 blog post
+
+For information of
+
+. Design of the plugin
+. What steps this plugin follow
+. What was done in codeing Phase 1
+. How to run the plugin.
+
+Please see link:https://jenkins.io/blog/2018/06/15/simple-pull-request-plugin/[Phase 1 blog post]
+
+=== Implementations till now
+
+Alpha version of the plugin is released. It supports all features of Multi-Branch Pipeline and offers the following features.
+
+Build description is defined via YAML file stored within the SCM repo. This plugin
+will depend on GitHub plugin, Bitbucket plugin, Gitlab plugin if users will be
+using respective paltfroms for their repositories.
+
+. Conversion of YAML to Declarative Pipeline: A class YamlToPipeline
+is written which will load the "Jenkinsfile.yaml" and make use of PipelineSnippetGenerator class
+to generate Declarative pipeline code.
+. Reporting of results, only xml report types is supprted for now.
+. Use of Yaml file (Jenkinsfile.yaml) from target branch.
+. Git Push step: To push the changes of pull request to the target branch. This is implemented
+using git-plugin, PushCommand is used for this from git-plugin. credentialId,
+branch name and repository url for intracting with Github, Bitbucket, etc
+will be taken automatically from "Branch-Source" (Users have to fill thes
+details of branch source in job configuration UI). (You can see
+link:https://github.com/Jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo])
+. StepConfigurator: To generate pipeline code for all supported steps in Jenkins. This is using
+Jenkins configuration-as-code plugin (JCasC plugin) to configure a particular step object and
+then that step object is passed to `Snippetizer.object2Groovy()` method to generate the script of that step.
+
+=== Jenkinsfile.yaml example
+
+For the phase 1 prototype demonstration, the following yaml file was used.
+Note that this format is subject to change in the next phases of the project,
+as we formalise the yaml format definition.
+[source, yaml]
+----
+#  Docket image agent example
+agent:
+ label: my_label
+ customWorkspace: path_to_workspace
+ dockerImage: maven:3-alpine
+ args: -v /tmp:/tmp
+
+  tools:
+    maven : maven_3.0.1
+    jdk : jdk8
+
+configuration:
+  # Push PR changes to the target branch if the build will be success.
+  # default value will be false
+  pushPrOnSuccess: false
+
+  prApprovers:
+    - username1
+    - username2
+    - username3
+
+# Study about the reports type. Urgent
+reports:
+  - location_of_report_1
+  - location_of_report_2
```

Comment:
This short section on reports should be commented out because it is not realistic. In the comment say that this section will be supported in phase 3.